types/codemirror/index.d.ts

```diff
@@ -945,13 +945,18 @@ declare namespace CodeMirror {
         clientHeight: any;
     }
 
-    interface TextMarker extends Partial<TextMarkerOptions> {
+    interface MarkerRange {
+        from: CodeMirror.Position;
+        to: CodeMirror.Position;
+    }
+
+    interface TextMarker<T> extends Partial<TextMarkerOptions> {
```

Comment:
You probably want to give a default type argument to avoid breaking people using the old form.

Reply:
Good idea (I didn't remember default type arguments were a thing)!
Since both `markText()` and `setBookmark()` are non-addon-methods, and I don't see any strong sign that one is used more often, I think the only reasonable default is `MarkerRange | Position`. That will probably still cause breakage for most people using `TextMarker` explicitly (after all, `find()`ing them is what they're here for, and `MarkerRange` and `Position` don't even share a single field), but I'd guess that's unavoidable.